draft-irtf-cfrg-hpke.md

```diff
@@ -420,34 +420,23 @@ and {{encryption-context}} for details.
 Suppose we are given a KDF, and a Diffie-Hellman group providing the
 following operations:
 
-- `GenerateKeyPair()`: Randomized algorithm to generate a key pair `(skX, pkX)`
-  for the DH group in use
-- `DeriveKeyPair(ikm)`: Deterministic algorithm to derive a key pair
-  `(skX, pkX)` from the byte string `ikm`, where `ikm` SHOULD have at
-  least `Nsk` bytes of entropy (see {{derive-key-pair}} for discussion)
-- `DH(skX, pkY)`: Perform a non-interactive DH exchange using the
-  private key `skX` and public key `pkY` to produce a Diffie-Hellman
-  shared secret of length `Ndh`. This function can raise a
-  `ValidationError` as described in {{validation}}.
-- `Serialize(pk)`: Produce a byte string of length `Npk`
-  encoding the public key `pk`
-- `Deserialize(enc)`: Parse a byte string of length `Npk` to recover a
-  public key. This function can raise a `DeserializeError` error upon
-  `enc` deserialization failure.
+- `DH(skX, pkY)`: Perform a non-interactive Diffie-Hellman exchange using
+  the private key `skX` and public key `pkY` to produce a shared secret of
+  length `Ndh`. This function can raise a `ValidationError` as described
+  in {{validation}}.
 - `Ndh`: The length in bytes of a Diffie-Hellman shared secret produced
-  by `DH()`
-- `Nsk`: The length in bytes of a Diffie-Hellman private key
-
-Since an encapsulated key is a Diffie-Hellman public key in this KEM
-algorithm, we use `Serialize()` to encode them, and `Npk` equals `Nenc`.
-The same applies to `Deserialize()`.
+  by `DH()`.
+- `Nsk`: The length in bytes of a Diffie-Hellman private key.
 
-Then we can construct a KEM called `DHKEM(Group, KDF)` in the
-following way, where `Group` denotes the Diffie-Hellman group and
-`KDF` the KDF. The function parameters `pkR` and `pkS` are deserialized
-public keys, and `enc` is a serialized public key.
-{{derive-key-pair}} contains the `DeriveKeyPair` function specification
-for DHKEMs defined in this document.
+Then we can construct a KEM that implements the interface defined in {{base-crypto}}
+called `DHKEM(Group, KDF)` in the following way, where `Group` denotes the
+Diffie-Hellman group and `KDF` the KDF. The function parameters `pkR` and `pkS`
+are deserialized public keys, and `enc` is a serialized public key. Since
+encapsulated keys are Diffie-Hellman public keys in this KEM algorithm,
+we use `SerializePublicKey()` and `DeserializePublicKey()` to encode and decode
```

Comment:
I like how the proposal reduced text repetition. One nit:

In the pseudo-code we still use `Serialize` and `Deserialize`, that are no longer defined. Switch to `SerializePublicKey()` and `DeserializePublicKey()` in the pseudo-code?

Reply:
Fixed and pushed.